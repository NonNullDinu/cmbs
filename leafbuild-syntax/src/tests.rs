fn parses_without_errors(s: &str) {
    assert_eq!(crate::parser::parse(s).errors, vec![]);
}

// Those are only *parsing* tests, so as long as they make sense
// *syntactically*, they should parse.
// Even if semantically they are void of meaning.
#[test]
fn unit_to_unit_assignment() {
    parses_without_errors("() = ()\n")
}

#[test]
fn line_trailing_space() {
    parses_without_errors("() = () \n")
}

#[test]
fn if_condition() {
    parses_without_errors("if (()) {} else {}")
}

#[test]
fn if_else_if_condition() {
    parses_without_errors("if () {} else if () {} else {}")
}

#[test]
fn if_else_if_else_condition_stretched() {
    parses_without_errors(
        r#"
    if ()

    {



    }

    else if () {


    }

    else {


    }
    "#,
    )
}

#[test]
fn if_condition_in_expr() {
    parses_without_errors("() = if () {} else {} \n")
}

#[test]
fn if_condition_in_expr_stretched() {
    parses_without_errors(
        r#"
    () = if ()
    {

    }

    else
    {


    }
    "#,
    )
}

#[test]
fn declaration() {
    parses_without_errors("let x = ()\n")
}

#[test]
fn foreach_basic() {
    parses_without_errors("foreach () in () {}")
}
