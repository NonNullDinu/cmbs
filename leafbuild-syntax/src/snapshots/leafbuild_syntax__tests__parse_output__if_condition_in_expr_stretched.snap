---
source: leafbuild-syntax/src/tests.rs
expression: "\n        () = if ()\n        {\n    \n        }\n    \n        else\n        {\n    \n    \n        }\n    "

---
- ROOT 0..97
  - "\n" NEWLINE 0..1
  - "        " WHITESPACE 1..9
  - Assignment 9..93
    - Expr 9..11
      - PrimaryExpr 9..11
        - TupleExpr 9..11
          - "(" L_PAREN 9..10
          - ")" R_PAREN 10..11
    - " " WHITESPACE 11..12
    - "=" EQ 12..13
    - " " WHITESPACE 13..14
    - Expr 14..92
      - PrimaryExpr 14..92
        - Conditional 14..92
          - ConditionalBranch 14..44
            - "if" IF_KW 14..16
            - " " WHITESPACE 16..17
            - Expr 17..19
              - PrimaryExpr 17..19
                - TupleExpr 17..19
                  - "(" L_PAREN 17..18
                  - ")" R_PAREN 18..19
            - "\n" NEWLINE 19..20
            - "        " WHITESPACE 20..28
            - ExprBlock 28..44
              - "{" L_BRACE 28..29
              - "\n" NEWLINE 29..30
              - "    " WHITESPACE 30..34
              - "\n" NEWLINE 34..35
              - "        " WHITESPACE 35..43
              - "}" R_BRACE 43..44
          - "\n" NEWLINE 44..45
          - "    " WHITESPACE 45..49
          - "\n" NEWLINE 49..50
          - "        " WHITESPACE 50..58
          - "else" ELSE_KW 58..62
          - "\n" NEWLINE 62..63
          - "        " WHITESPACE 63..71
          - ExprBlock 71..92
            - "{" L_BRACE 71..72
            - "\n" NEWLINE 72..73
            - "    " WHITESPACE 73..77
            - "\n" NEWLINE 77..78
            - "    " WHITESPACE 78..82
            - "\n" NEWLINE 82..83
            - "        " WHITESPACE 83..91
            - "}" R_BRACE 91..92
    - "\n" NEWLINE 92..93
  - "    " WHITESPACE 93..97

